File: android/src/main/kotlin/com/pvs/flutter_utils/ads/NativeVideoViewFactory.java
package com.pvs.flutter_utils.ads;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Map;

import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class NativeVideoViewFactory extends PlatformViewFactory {

    String videoid;
   public NativeVideoViewFactory(String nativeId) {
        super(StandardMessageCodec.INSTANCE);
        videoid = nativeId;
    }
    @SuppressWarnings("unchecked")
    @NonNull
    @Override
    public PlatformView create(@NonNull Context context, int id, @Nullable Object args) {
        final Map<String, Object> creationParams = (Map<String, Object>) args;

        return new NativeVideoView(context, id, creationParams,videoid);
    }
}

File: android/src/main/kotlin/com/pvs/flutter_utils/NativeVideoViewFactory.kt
package com.example.flutter_native_ad

import android.content.Context
import java.util.Map
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeVideoViewFactory internal constructor(var videoid: String?) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    @NonNull
    @Override
    fun create(@NonNull context: Context?, id: Int, @Nullable args: Object?): PlatformView? {
        val creationParams: Map<String?, Object?>? = args as Map<String?, Object?>?

        return NativeVideoView(context, id, creationParams, videoid)
    }
}
